### PR TITLE
Fix memory initialization errors and more general path

### PR DESCRIPTION
**Pull request by TooAngel**
Small fixes for the `stats.add(` method to be more generic.
Fix initializiation errors for enabling / disabling stats
and `delete Memory.stats`

### CODE REVIEW

**Comment by TooAngel** [comment]
DCBOT: 
This repository is under [democratic collaboration](https://github.com/TooAngel/democratic-collaboration) and will be merged automatically.

The merge decision is based on the outcome of the reviews:
 - `Approve` add the reviewer value (number of commits) to the `metric`
 - `Request changes` substract the reviewer value from the `metric`

 The merge will happen after `(1 - metric/total.votes) * (5 + pull_request.commits * 5))` calculated from the last code change.

Please review the PR to make a good democratic decision.

Summoning some reviewers:
 - @Nevrdid: I always like your reviews
 - @WhiteHalmos: A proper review is appreciated
 - @Somotaw: I just want to get this merged ASAP

File: src/prototype_room_creepbuilder.js [hunk 1]
```diff
@@ -51,8 +50,10 @@ Room.prototype.spawnCreateCreep = function(role, heal, level, squad, routing, ba
...
       continue;
     }
-    let userName = Memory.username || _.find(Game.spawns, 'owner').owner;
     if (config.stats.enabled) {
+      let userName = Memory.username || _.find(Game.spawns, 'owner').owner;
+      Memory.stats = Memory.stats || {};
+      Memory.stats[userName].roles = Memory.stats[userName].roles || {};
       let roleStat = Memory.stats[userName].roles[role];
       let previousAmount = roleStat ? roleStat : 0;
       Memory.stats[userName].roles[role] = previousAmount + 1;
```
Comment (Nevrdid) [inline src/prototype_room_creepbuilder.js]: `brain.stats.add(['roles', role], previousAmount + 1);`
could work so i supose and we do no more need name define and previous init i think.
there is the decrease one in brain_memory l.55. you could change it the same i believe.

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 99/290
    Coefficient: 0.341379310345
    Merging in 4 days -7 hours
    Age 2 days 21 hours

File: src/prototype_room_creepbuilder.js (hunk 1 above)
Comment (TooAngel) [inline src/prototype_room_creepbuilder.js]: Tried, I can't.

To get the `previousAmount` you need roleStat. To make sure this works you need to make sure stats are available. So you need the username.

Comment (Nevrdid) [inline src/prototype_room_creepbuilder.js]: 👍 ^^' forget

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 197/290
    Coefficient: 0.679310344828
    Merging in 1 days -17 hours
    Age 2 days 21 hours

**Pull request merged**